agentops/session.py

```diff
@@ -43,21 +46,52 @@ def set_session_video(self, video: str) -> None:
         """
         self.video = video
 
+    def add_metadata(self, key: str, value: Any) -> None:
+        """
+        Adds metadata to the session.
+
+        Args:
+            key (str): The metadata key.
+            value (Any): The metadata value.
+        """
+        if self.metadata is None:
+            self.metadata = {}
+        self.metadata[key] = value
+
     def end_session(
         self, end_state: str = "Indeterminate", end_state_reason: Optional[str] = None
     ) -> None:
         """
-        End the session with a specified state, rating, and reason.
+        End the session with a specified state and reason.
 
         Args:
             end_state (str, optional): The final state of the session. Options: "Success", "Fail", "Indeterminate"
-            rating (str, optional): The rating for the session.
             end_state_reason (str, optional): The reason for ending the session. Provides context for why the session ended.
         """
         self.end_state = end_state
         self.end_state_reason = end_state_reason
         self.end_timestamp = get_ISO_time()
 
+    def expire_session(self) -> None:
```

Comment:
I dont see anywhere in this PR that the function is being used. implementing it to automatically expire may be more effort than it's worth and would likely require a dedicated thread per session. I'm not sure if we want to make the SDK more processor intensive with this. I'm open to other options though if you have others in mind!